src/main/scala/ch/epfl/bluebrain/nexus/admin/organizations/Organizations.scala

```diff
@@ -1,16 +1,140 @@
 package ch.epfl.bluebrain.nexus.admin.organizations
 
+import java.time.Clock
 import java.util.UUID
 
-import ch.epfl.bluebrain.nexus.admin.types.ResourceMetadata
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import cats.MonadError
+import cats.effect.ConcurrentEffect
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig.HttpConfig
+import ch.epfl.bluebrain.nexus.admin.exceptions.UnexpectedState
+import ch.epfl.bluebrain.nexus.admin.index.Index
+import ch.epfl.bluebrain.nexus.admin.organizations.OrganizationCommand._
+import ch.epfl.bluebrain.nexus.admin.organizations.OrganizationState._
+import ch.epfl.bluebrain.nexus.admin.types.ResourceF
 import ch.epfl.bluebrain.nexus.commons.types.identity.Identity
+import ch.epfl.bluebrain.nexus.sourcing.Aggregate
+import ch.epfl.bluebrain.nexus.sourcing.akka.{AkkaAggregate, AkkaSourcingConfig, PassivationStrategy, RetryStrategy}
 
-class Organizations[F[_]] {
+/**
+  * Organizations operations bundle
+  */
+class Organizations[F[_]](agg: Agg[F], index: Index)(implicit F: MonadError[F, Throwable],
+                                                     clock: Clock,
+                                                     http: HttpConfig) {
 
-  type OrganizationMetaOrRejection = Either[OrganizationRejection, ResourceMetadata]
+  /**
+    * Create an organization.
+    *
+    * @param organization organization to create
+    * @param caller       identity of the caller performing the operation
+    * @return             metadata about the organization
+    */
+  def create(organization: Organization)(implicit caller: Identity): F[OrganizationMetaOrRejection] =
+    index.getOrganization(organization.label) match {
+      case None    => evaluate(CreateOrganization(UUID.randomUUID(), organization, rev = 0L, clock.instant(), caller))
+      case Some(_) => F.pure(Left(OrganizationAlreadyExists))
+    }
 
-  def create(organization: Organization)(implicit caller: Identity): F[OrganizationMetaOrRejection]            = ???
-  def update(organization: Organization, rev: Long)(implicit caller: Identity): F[OrganizationMetaOrRejection] = ???
-  def deprecate(id: UUID, rev: Long)(implicit caller: Identity): F[OrganizationMetaOrRejection]                = ???
+  /**
+    * Update an organization.
+    *
+    * @param label        label of the organization to update
+    * @param organization the updated organization
+    * @param rev          the latest known revision
+    * @param caller       identity of the caller performing the operation
+    * @return
+    */
+  def update(label: String, organization: Organization, rev: Long)(
+      implicit caller: Identity): F[OrganizationMetaOrRejection] =
+    index.getOrganization(label) match {
```

Comment:
I would recommend using the public function `resolve` that returns an `id` for a `label` and call that one instead of reading off the index every time.

Reply:
is it possible to update the label of an organization?

Reply:
> I would recommend using the public function resolve that returns an id for a label and call that one instead of reading off the index every time.
 
the `resolve` function would have to read off the index as well?

> is it possible to update the label of an organization?

yes, that's why we're not using labels as ids. Not sure how that will work on the HTTP routes level yet.

Reply:
yes